.github/workflows/CI.yml

```diff
@@ -58,6 +64,30 @@ jobs:
           else
             cargo test-amaru --locked --target ${{ matrix.environments.target }}
           fi
+        shell: bash
+  build-crates:
+    name: Build crates on ${{ matrix.environments.runner }} with target ${{ matrix.environments.target }}
+    needs: sanity
+    strategy:
+      fail-fast: false
+      matrix:
+        environments:
+          - runner: ubuntu-latest
+            target: wasm32-unknown-unknown
+            package: amaru-ledger
+            optional: true
+    timeout-minutes: 30
+    runs-on: ${{ matrix.environments.runner }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: ${{ runner.os }}-${{ matrix.environments.target }}
+      - run: rustup target add ${{ matrix.environments.target }}
+      - name: Run tests
+        run: |
+          set +e
+          cargo test --locked --all-features -p ${{ matrix.environments.package }} --target ${{ matrix.environments.target }}
```

Comment:
I am not sure I follow here? Why build the ledger separately from the rest (and has a separate job altogether)? Couldn't the same be achieved by adding packages to the matrix in the previous job? And more importantly, would the previous job _already build and test_ the ledger crate?

Reply:
`ledger` would indeed be tested by the previous job but some target (e.g. `wasm`) won't realistically compile the full `amaru`, failing the whole job. So the idea is to restrict to some identified packages those specific targets.

I can see if there is some way to combine both.